languages/go/reference/exercise-concepts/scrabble.md

```diff
@@ -0,0 +1,34 @@
+# Concepts of Scrabble
+
+There are different approaches to scrabble which are in the different categories below. Common concepts for all approaches are in general.
+
+## General
+- zero value variable declaration: The `score` variable for the total score needs initialization to `0`.
+- strings: lower and upper case letters should be treated equally (`ToUpper`/`ToLower`).
+- range loop over string: using a `range` loop on a string is a bit special. Its index counts bytes, its value is a rune.
+- for loop: This exercise can also be solved with a `for loop` -- although `range` is preferable here.
+- runes: using a range loop over a string will return a rune.
+- bytes: using a for loop comes with indexing into a string which returns a `byte`.
+- unicode: instead of the entire string (`strings`) a single `rune` can be transformed `ToUpper`/`ToLower`.
+- numbers: adding a value to an existing sum.
+- comments: comment on exported function should be present. Best also on package itself, but that is more optional.
+- ignore values: In the `range` loop the index is not needed and has to be ignored with a `_` (underscore).
+
+## Approach: Switch
+
+- switch: uses switch to get the score per letter
+- functions: makes the code nicer to extract the big switch
+
+## Aproach: map
+
+- map: is used to store the score per letter. Map lookup. Map definition with content.
```

Comment:
with a map defined using string, and trying to lookup with rune: conversion between types ?

Reply:
defining map outside function will ensure it is only called once

Reply:
@hans-d: map should be `map[run]int`

Reply:
But of course some might use type conversion. I'll add it